src/components/Editable/useEditMode.ts

```diff
@@ -136,6 +138,18 @@ const useEditMode = ({
     })
   }, [])
 
+  // Resume focus if sidebar was just closed and isEditing is true
+  useEffect(() => {
+    if (showSidebar) {
+      setHadSidebar(true)
+    } else {
+      if (isEditing && hadSidebar) {
+        contentRef.current?.focus()
+      }
+      setHadSidebar(false)
```

Comment:
Could the two `setHadSidebar` statements be simplified into one?

```
setHadSidebar(showSidebar);
```

---

Additionally, this effect hook gets triggered twice because `hadSidebar` is a state value. It gets updated within the hook, then triggers it a second time because it has updated. Could we change it to `const hadSidebar = useRef(false)` and update it with `hadSidebar.current = showSidebar` at the end of the effect? Then, it can be removed as a dependency and only trigger the effect hook once without breaking the functionality.

---

Lastly, there is another PR proposing a `usePrevious` hook. I don't think refactoring this PR to make use of a `usePrevious` hook is necessary, but it would be nicer (and more legible) to simply have:

```
const showSidebar = useSelector(state => state.showSidebar)
const hadSidebar = usePrevious(showSidebar)
```